### ISSUE TITLE
[Tweets] Red Won't Exit for Restart/Shutdown

### ISSUE THREAD

**Issue by Menteroth** [TrustyJAID/Trusty-cogs#83]
For some reason your `tweets` cog is making shutdown/restart command not work. Red freezes and won't exit to command prompt. I know it's this cog because I loaded and tried all of my cogs until it happened again. I also tried `--no-cogs` and only loaded `tweets`.

Using either BAT file or `redbot <instance>` to start. Same with both.

**Comment by TrustyJAID** [comment]
I'll need more information about this to help you solve it. What OS are you running? Are you on Dev or current Red? Are there any logs or tracebacks? Are you using the current version of tweepy? Any and all information you can give me will help me figure out how to solve this for you. I have seen the issue before but have made arrangements it fix it in the past so I'm not quite sure what's going wrong for you.

**Comment by Menteroth** [comment]
Forgot to mention I have used it for quite some time, it was just like a week ago it happened. Not sure if it was when I updated the cog or not. But updated just before this comment and tried, still same.

Yeah, sorry for the lack of info in the initial post.

```
Windows 7 64-bit.
Red 3.1.8
From lib folder: tweepy-3.8.0.dist-info
```

This is all I get in the console before it freezes.
```
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x0000000014CFB8E8>, 12517784.022)]', '[(<aiohttp.client_proto.ResponseHandler object
 at 0x00000000055F6D08>, 12517784.162)]', '[(<aiohttp.client_proto.ResponseHandler object at 0x0000000008C356A8>, 12517784.365)]', '[(<aiohttp.client_
proto.ResponseHandler object at 0x000000000411B1C8>, 12517784.927)]', '[(<aiohttp.client_proto.ResponseHandler object at 0x0000000007C7D6A8>, 12517785
.488)]', '[(<aiohttp.client_proto.ResponseHandler object at 0x0000000015857768>, 12517786.003)]', '[(<aiohttp.client_proto.ResponseHandler object at 0
x00000000141AB648>, 12517786.221)]']
connector: <aiohttp.connector.TCPConnector object at 0x00000000083B9508>

_GatheringFuture exception was never retrieved
future: <_GatheringFuture finished exception=CancelledError()>
concurrent.futures._base.CancelledError
```

**Comment by Menteroth** [comment]
This happened during the day also. 

```
Exception in thread Thread-10:
Traceback (most recent call last):
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 554, in _get_chunk_left
    chunk_left = self._read_next_chunk_size()
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 521, in _read_next_chunk_size
    return int(line, 16)
ValueError: invalid literal for int() with base 16: b''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 586, in _readinto_chunked
    chunk_left = self._get_chunk_left()
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 556, in _get_chunk_left
    raise IncompleteRead(b'')
http.client.IncompleteRead: IncompleteRead(0 bytes read)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\urllib3\response.py", line 425, in _error_catcher
    yield
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\urllib3\response.py", line 507, in read
    data = self._fp.read(amt) if not fp_closed else b""
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 457, in read
    n = self.readinto(b)
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 491, in readinto
    return self._readinto_chunked(b)
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\http\client.py", line 602, in _readinto_chunked
    raise IncompleteRead(bytes(b[0:total_bytes]))
http.client.IncompleteRead: IncompleteRead(720 bytes read)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\tweepy\streaming.py", line 320, in _run
    six.reraise(*exc_info)
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\six.py", line 696, in reraise
    raise value
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\tweepy\streaming.py", line 289, in _run
    self._read_loop(resp)
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\tweepy\streaming.py", line 339, in _read_loop
    line = buf.read_line()
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\tweepy\streaming.py", line 200, in read_line
    self._buffer += self._stream.read(self._chunk_size)
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\urllib3\response.py", line 529, in read
    raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
  File "C:\Users\Chris\AppData\Local\Programs\Python\Python37\lib\contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "C:\__\coding\Soundwave\cogs\Downloader\lib\urllib3\response.py", line 443, in _error_catcher
    raise ProtocolError("Connection broken: %r" % e, e)
urllib3.exceptions.ProtocolError: ('Connection broken: IncompleteRead(720 bytes read)', IncompleteRead(720 bytes read))
```

**Comment by Menteroth** [comment]
That last post with errors I made is a repeating thing. Just thought I'd add that info as well.

**Comment by mikeshardmind** [comment]
This is *probably* fixed upstream by the updates to python 3.8.1 and red 3.2.1 as this issue was a motivating factor in the python version requirement update.

**Comment by TrustyJAID** [comment]
This should properly be fixed now in 2.5.2 [ebadc7cfde25f3868c8874563fa0b1b20a4ab292](https://github.com/TrustyJAID/Trusty-cogs/commit/ebadc7cfde25f3868c8874563fa0b1b20a4ab292)
